src/Infrastructure/Services/UserService.cs

```diff
@@ -70,11 +77,13 @@ public class UserService : IUserService
             _emailService = emailService;
             _nhsLoginHttpClient = nhsLoginHttpClient;
             _mediator = mediator;
-            _devSettings = devSettings.Value;
+            _devSettings = devSettings;
             _participantService = participantService;
             _dataProtector = dataProtector.CreateProtector("mfa.login.details");
             _contentfulService = contentfulService;
             _contentfulSettings = contentfulSettings;
+            _mockIdentityService = mockIdentityService;
+            _environment = environment;
         }
 
         private string GenerateMfaDetails(AdminInitiateAuthResponse response, string password = null)
```

Comment:
This is the method that has different behaviour depending on settings / environment.

This should be a separate interface with production / development implementations configured on startup.